### ISSUE TITLE
Change or let option to chnage log level attribute to upper case

### ISSUE THREAD

**Issue by AMontagu** [grafana/faro-web-sdk#194]
## Description

Following my understanding of open telemetry specification default log level is uppercase (but I only understand this based on default value when only assign default SeverityNumber so I can be wrong).
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-severitynumber

Actually what is sure is by default python logging level is in uppercase. So when receiving the log from both front end and back end I have both uppercase and lowercase.

This is of course not a big deal as we can make case insensitive query in grafana agent but it add some complexity that may in some langage be harder to handle if the mapping is not as easy as case insensitive.

## Proposed solution

Add a custom mapping option that map the log level to the log level we want to sent to grafana agent.
https://github.com/grafana/faro-web-sdk/blob/700297cfd76792b1c0358eb76d8cb864a4badde9/packages/web-sdk/src/instrumentations/console/instrumentation.ts#L24

Can make an MR if it's something you are agree with but not in the project priorities.

**Comment by codecapitano** [comment]
Hey @AMontagu thank you for opening the issue and even more for your offer to open a PR 🥳 

An option to already solve your goal is to use the [`beforeSend` hook](https://grafana.com/docs/grafana-cloud/frontend-observability/faro-web-sdk/faq/#does-the-faro-sdk-have-a-sampling-capability-to-rate-limit-the-telemetry-data-sent-to-grafana-cloud) to change the casing of the log level. 

For example:

```ts
initializeFaro({
  // ... faro configuration
  beforeSend: (item) => {
    if (item.type === 'log') {
      return {...item, level: item.level.toUppercase()}
    }
  }
```

Independent of that we will discuss your idea and I'll get back to you if like to wait a bit?

Note:
The internal Faro model is build upon the OTel spec. 
In case an otlp compliant transport is needed we are already working on that.
You c an find it in the `experimental` package but beware, the agent doesnlt support this payload yet.

Many thanks for all the testing, input and help to drive the Web-SDK forward
Cheers, Marco

**Comment by AMontagu** [comment]
Thank you for the fast reply.
Will use the beforeSend.

What do you mean by "doesn't support this payload yet" ?

Can I help by testing it or anything else ?

With the workaround I can wait all the time you want and If you want a PR tell me and I will do.

**Comment by codecapitano** [comment]
> What do you mean by "doesn't support this payload yet" ?
It's a different model than the Faro one and the receiver needs to be updated to handle this as well.

> Can I help by testing it or anything else ?
Thanks a lot and great that you are so much into the web-sdk 🙏  

If you use the Grafana agent you can not really use the OtlpHttpTransport.

But there are other interesting experimental packages like the auto instrumentation of the Performance Timeline of the browser. In case you're interested in that kind of stuff.

> With the workaround I can wait all the time you want and If you want a PR tell me and I will do. 
Awesome and glad `beforeSend`works for you 

Cheers,
Marco

**Comment by AMontagu** [comment]
@codecapitano is that the beforeSend the recommend validated way ?

**Comment by codecapitano** [comment]
Yes, for now we don't have any plans to implement a dedicated logic to change case of range names.
